### PR TITLE
Added viewportPercentile() function

### PR DESCRIPTION
**Pull request by davidmanzanares**
Partial work for https://github.com/CartoDB/renderer-prototype/issues/19

### CODE REVIEW

File: src/core/style/expressions/quantiles.js [hunk 1]
```diff
@@ -54,6 +54,8 @@ export default class Quantiles extends Expression {
         const total = column.accumHistogram[999];
         const r = Math.random();
         let brs = [];
+
+        // TODO OPT: this could be faster with binary search
         this.breakpoints.map((breakpoint, index) => {
             for (i; i < 1000; i++) {
```
Comment (IagoLast) [inline src/core/style/expressions/quantiles.js]: why 1000? 

https://refactoring.guru/replace-magic-number-with-symbolic-constant

File: src/core/style/expressions/viewportAggregation.js [hunk 2]
```diff
@@ -39,3 +39,48 @@ function generateViewportExpression(metadataPropertyName) {
         }
     };
 }
+
+
+export class ViewportPercentile extends Expression {
```
Comment (IagoLast) [inline src/core/style/expressions/viewportAggregation.js]: should this be moved to its own file?

File: src/core/style/expressions/viewportAggregation.js [hunk 3]
```diff
@@ -39,3 +39,48 @@ function generateViewportExpression(metadataPropertyName) {
...
+        return this.property._getMinimumNeededSchema();
+    }
+    _getDrawMetadataRequirements() {
+        return { columns: [this.property.name] };
+    }
+    _preDraw(drawMetadata, gl) {
+        const column = drawMetadata.columns.find(c => c.name == this.property.name);
+        const total = column.accumHistogram[999];
+        // TODO OPT: this could be faster with binary search
+        for (var i = 0; i < 1000; i++) {
```
Comment (IagoLast) [inline src/core/style/expressions/viewportAggregation.js]: [https://github.com/CartoDB/renderer-prototype/pull/65/files#r168210011](same)

File: src/core/style/expressions/viewportAggregation.js [hunk 4]
```diff
@@ -39,3 +39,48 @@ function generateViewportExpression(metadataPropertyName) {
...
+        }
+        const br = i / 1000 * (column.max - column.min) + column.min;
+        this.value.expr = br;
+
+        if (Math.random() > 0.99) {
+            console.log(`percentile${this.percentile}`, this.property.name, br, drawMetadata);
+        }
+        this.value._preDraw(drawMetadata, gl);
+    }
+}
```
Comment (IagoLast) [inline src/core/style/expressions/viewportAggregation.js]: end files with newline `.editorconfig` can take care of this

File: src/core/style/expressions/quantiles.js (hunk 1 above)
Comment (davidmanzanares) [reply src/core/style/expressions/quantiles.js]: Yep, no need for the link :slightly_smiling_face:

File: src/core/style/expressions/viewportAggregation.js [hunk 5]
```diff
@@ -39,3 +39,48 @@ function generateViewportExpression(metadataPropertyName) {
...
+    _preDraw(drawMetadata, gl) {
+        const column = drawMetadata.columns.find(c => c.name == this.property.name);
+        const total = column.accumHistogram[999];
+        // TODO OPT: this could be faster with binary search
+        for (var i = 0; i < 1000; i++) {
+            if (column.accumHistogram[i] >= this.percentile / 100 * total) {
+                break;
+            }
+        }
+        const br = i / 1000 * (column.max - column.min) + column.min;
```
Comment (IagoLast) [inline src/core/style/expressions/viewportAggregation.js]: br?

File: src/core/style/expressions/viewportAggregation.js (hunk 2 above)
Comment (davidmanzanares) [reply src/core/style/expressions/viewportAggregation.js]: I don't think so, it is just a specialization of the other viewport aggregation functions.

File: src/core/style/expressions/viewportAggregation.js (hunk 4 above)
Comment (davidmanzanares) [reply src/core/style/expressions/viewportAggregation.js]: :+1:

File: src/core/style/expressions/viewportAggregation.js [hunk 6]
```diff
@@ -39,3 +39,48 @@ function generateViewportExpression(metadataPropertyName) {
...
+        this.value.expr = br;
+
+        if (Math.random() > 0.99) {
```
Comment (IagoLast) [inline src/core/style/expressions/viewportAggregation.js]: are we going to leave this in production?

File: src/core/style/expressions/viewportAggregation.js (hunk 5 above)
Comment (davidmanzanares) [reply src/core/style/expressions/viewportAggregation.js]: :+1: Will change

File: src/core/style/expressions/viewportAggregation.js (hunk 2 above)
Comment (IagoLast) [reply src/core/style/expressions/viewportAggregation.js]: 🤔 what about creating a viewportAggregation folder with files in it?

File: src/core/style/expressions/viewportAggregation.js (hunk 6 above)
Comment (davidmanzanares) [reply src/core/style/expressions/viewportAggregation.js]: I'd leave it for now, there are other console logs that must be removed at some point, so...

File: src/core/style/expressions/viewportAggregation.js (hunk 2 above)
Comment (davidmanzanares) [reply src/core/style/expressions/viewportAggregation.js]: Too much noise for a 86 lines long file. And this pattern of having multiple classes of the same type like binary operations are in other places too, with a similar, if not less, line count.

**Pull request merged**